packages/authorization-process/test/createKeys.test.ts

```diff
@@ -0,0 +1,323 @@
+/* eslint-disable functional/no-let */
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import crypto from "crypto";
+import { describe, it, vi, beforeAll, afterAll, expect } from "vitest";
+import {
+  Client,
+  ClientKeyAddedV2,
+  Key,
+  TenantId,
+  UserId,
+  generateId,
+  notAllowedPrivateKeyException,
+  toClientV2,
+} from "pagopa-interop-models";
+import { AuthData, genericLogger } from "pagopa-interop-commons";
+import {
+  decodeProtobufPayload,
+  getMockKey,
+  readLastEventByStreamId,
+} from "pagopa-interop-commons-test/index.js";
+import { getMockClient } from "pagopa-interop-commons-test";
+import { selfcareV2Client } from "pagopa-interop-selfcare-v2-client";
+import { ApiKeySeed, ApiKeysSeed } from "../src/model/domain/models.js";
+import {
+  clientNotFound,
+  keyAlreadyExists,
+  organizationNotAllowedOnClient,
+  securityUserNotFound,
+  tooManyKeysPerClient,
+  userNotFound,
+} from "../src/model/domain/errors.js";
+import {
+  calculateKid,
+  createJWK,
+  decodeBase64ToPem,
+} from "../../commons/src/auth/jwk.js";
+import { addOneClient, authorizationService, postgresDB } from "./utils.js";
+
+describe("createKeys", () => {
+  const consumerId: TenantId = generateId();
+  const userId: UserId = generateId();
+
+  beforeAll(async () => {
+    vi.useFakeTimers();
+    vi.setSystemTime(new Date());
+  });
+
+  afterAll(() => {
+    vi.useRealTimers();
+  });
+
+  const key = crypto.generateKeyPairSync("rsa", {
+    modulusLength: 2048,
+  }).publicKey;
+
+  const pemKey = Buffer.from(
+    key.export({ type: "pkcs1", format: "pem" })
+  ).toString("base64");
+
+  const keySeed: ApiKeySeed = {
+    name: "key seed",
+    use: "ENC",
+    key: pemKey,
+    alg: "",
+  };
+
+  const keysSeeds: ApiKeysSeed = [keySeed];
+
+  function mockSelfcareV2ClientCall(
+    value: Awaited<
+      ReturnType<typeof selfcareV2Client.getInstitutionProductUsersUsingGET>
+    >
+  ): void {
+    vi.spyOn(
+      selfcareV2Client,
+      "getInstitutionProductUsersUsingGET"
+    ).mockImplementationOnce(() => Promise.resolve(value));
+  }
+
+  const mockSelfCareUsers = {
+    id: generateId(),
+    name: "test",
+    roles: [],
+    email: "test@test.it",
+    surname: "surname_test",
+  };
+
+  const mockAuthData: AuthData = {
+    organizationId: consumerId,
+    selfcareId: generateId(),
+    externalId: {
+      value: "",
+      origin: "",
+    },
+    userId,
+    userRoles: [],
+  };
+
+  const mockClient: Client = {
+    ...getMockClient(),
+    users: [userId],
+    consumerId,
+  };
+
+  it("should create the keys and add it to client", async () => {
+    mockSelfcareV2ClientCall([mockSelfCareUsers]);
+
+    await addOneClient(mockClient);
+
+    vi.mock("pagopa-interop-selfcare-v2-client", () => ({
+      selfcareV2Client: {
+        getInstitutionProductUsersUsingGET: (): Promise<boolean> =>
+          Promise.resolve(true),
```

Comment:
shouldn't this function return an array?

Reply:
This is a fake function that allows you to pass the security user check in any case, we don't care about the type of return, as long as the promise is resolved

Reply:
I realized that function is useless, so I deleted it